tests/jenkins/README.md

```diff
@@ -0,0 +1,55 @@
+# Tests for Mozilla's Treeherder
```

Comment:
I think we can just delete this file - most of it is no longer relevant - and the advice for contributing doesn't really make sense since the Jenkins files are now feature-frozen.